api/v1/recorder/recorder.proto

```diff
@@ -49,6 +49,26 @@ message StartRecording {
     // max_capture_length specifies the maximum packet length.
     // Full packet length will be captured if absent/zero.
     uint32 max_capture_length = 3;
+
+    // stop_condition defines conditions which will cause the recording to
+    // stop early after any of the stop conditions has been hit
+    StopCondition stop_condition = 4;
+}
+
+// StopCondition defines one or more conditions which cause the recording to
+// stop after they have been hit. Stop conditions are ignored if they are
+// absent or zero-valued. If multiple conditions are defined, the recording
+// stops after the first one is hit.
+message StopCondition {
+    // bytes_captured stops the recording after at least this many bytes have
+    // been captured. Note: The resulting file might be slightly larger due
+    // to added pcap headers.
+    uint64 bytes_captured = 1;
+    // bytes_captured stops the recording after at least this many packets have
+    // been captured.
+    uint64 packets_captured = 2;
+    // nanoseconds_elapsed stops the recording after this duration has elapsed.
+    uint64 nanoseconds_elapsed = 3;
```

Comment:
Do we by chance want to pull in the protobuf duration helpers, like we do for time?

Reply:
Other Hubble code so far has used `nanoseconds`:
https://github.com/cilium/cilium/blob/6f9b87549f23effae024934ec3fcdec4dba95aa3/api/v1/flow/flow.proto#L159-L160


But let's give `Google.Protobuf.WellKnownTypes.Duration` a try. It would be the first use of it in Hubble, afaict (so far, we only used the timestamp). If it doesn't complicate the code, it's probably nicer.

Reply:
Using `google.protobuf.Duration` worked out nicely!